lib: location: Remove A-GNSS request fallback

tokangas opened the pull request:
Remove fallback to `NRF_MODEM_GNSS_EVT_AGNSS_REQ` event for determining GNSS assistance data need.
`nrf_modem_gnss_agnss_expiry_get()` can be used with all supported modem firmware versions.

Simplified GNSS assistance data handling.

Cleaned up error logging and some comments.

trantanen commented on lib/location/method_gnss.c:
```diff
@@ -1175,38 +1142,18 @@ static void method_gnss_agnss_expiry_process(const struct nrf_modem_gnss_agnss_e
-/* Queries assistance data need from GNSS.
- *
- * To maintain backward compatibility with older modem firmware versions, two different methods are
- * supported. If the nrf_modem_gnss_agnss_expiry_get() API is not supported (pre-v1.3.2 modem
```
I cannot add comment to old line 1258 but there are similar comments about if this is not supported etc.

trantanen commented on lib/location/method_gnss.c:
```diff
@@ -1175,38 +1142,18 @@ static void method_gnss_agnss_expiry_process(const struct nrf_modem_gnss_agnss_e
- * in case assistance data is currently needed.
- */
+/* Queries assistance data need from GNSS. */
 static void method_gnss_agnss_req_work_fn(struct k_work *work)
```
I'm thinking if this still needs to be running as a separate worker... Not sure but now that start and stop are not used, are all functions simply synchronous. Things would simplify and we might get rid of a sleep on line 1264.

Maybe function name could be something like `method_gnss_agnss_data_needed()` etc.

tokangas replied:
I think the extra sleep could be removed without making the function synchronous as well, but you are right, this can now be simplified further by making the A-GNSS data need check synchronous. I'll change it.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12815/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12815/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-nrf-iot_mosh | X | | |
| test-fw-nrfconnect-nrf-iot_positioning | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12815/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

tokangas commented:
I renamed `method_gnss_positioning_work_fn()` to `method_gnss_start_work_fn()` to align the work item and work handler naming, and moved a couple of functions to a different place. Because of this the diff looks worse than the changes really are.

trantanen reviewed (approved):
Good changes to make the code easier to read
